### ISSUE TITLE
Discussion: Redesigning Arthropods

### ISSUE THREAD

**Issue by Venera3** [CleverRaven/Cataclysm-DDA#45030]
# Is your feature request related to a problem? Please describe.
The game has a wide range of insects and assorted creepy crawlies which are universally way more boring than they should be. This comes from a few places, ranging from possible oversights ( all arthropods are blinder than a flashbanged bat ) to not utilizing later additions ( most of their special attacks are hardcoded, nothing uses the spell system) and generally bland flavor.

# Describe the solution you'd like
I'm planning on redoing all members of the spider/insect json and any other neglected monsters I find particularly strong about and/or figure out something cool for. This will all be JSON, but I'll make some notes below in case an actual programmer wants to help out.
Generally I'm aiming to give each species/group a lifecycle with at least one reproductive form to make the world feel a bit more alive and to make pacifying large swathes of land less trivial. If it makes sense I'll try to include predation/interspecies hostility, but aggroing is an all-or-nothing deal, so we'll have to see how that shakes out in game.

Concepts so far, with ideas that would require coding behind "//":

- Dragonflies: the naiads (larvae) are aquatic hunters with a reach attack, the adults are fast predators in medium-sized swarms, dangerous up until mid-late game with high armor piercing and a grab attack. Mega variant lays eggs, hits like a truck but can be outrun easily. //Some pathing tweaks to make them lay eggs in water

- Centipedes: small, frightened young that grow slowly into the most dangerous predator of the swamps with a deadly venom and high enough damage output to threaten endgame chars. Mega variant is a centipede mother, slightly slower from carrying her young which fall off and join the fight as it goes on. //A monster spell trigger at low health would make the spawning nicer

- Mosquitos: large swarms of hard to hit, low-damage annoyances to keep earlygame characters entertained and to give the predators something to hunt. Mega becomes a blood-gorged spiky ball that spawns smaller mosquitos (+maybe a ranged attack?).

- Bees: add queens and drones, queens lay eggs that have a small chance to spawn a drone and a miniscule chance for a new queen. Workers grow larger with age, but are relatively docile until you mess with them, they also drop wax. Drones have good vision, aggro readily and have a pheromone sting to "mark" you ( if scents work, will need some testing). Queens are mean in a fight, but the main difficulty should come from numbers and the venom. // Custom wandering hordes to simulate swarms and building new hives, monfaction-specific charm spells for Insect mutants to suborn them, intrahive fighting to simulate a nascent queen

- Wasps: egglaying queens+worker growth like in the bees, add a small chance for worker wasps to drop balls of flesh (randomly clean/mutated/tainted) to simulate them hunting. If dermatiks don't become flies they'd get a small chance to spawn from the eggs. Agressive from the get-go, camping out near a wasp nest should be suicidal for any non-lategame character. // Gametime swarming and nest building

- Flies: are pretty boring, so they also become mostly prey animals. Adding dermatiks into their evolution line might make sense from the resemblance to IRL botflies, but I find it hard to make them interesting. Maybe turn them into fireflies to make the swamps light up a bit.

- Spiders: since they are pretty similar anyway turn them into social hunters (some spider species form communities to cooperate in the hunt IRL). Common egg that can be laid by all and can hatch any of the group or common large egglayer evolution. Wolf spiders have good vision and are fast, jumping spiders get longer leaps and web spiders get a thrown web. Also make the spider pit actually spawn, make you scared to go in those webbed forests and maybe rehabilitate the trapdoor spider.

- Crayfish, slugs, other assorted forgotten animals: my knowledge of cool animal facts is pretty insect-centric, so for these I'll need some suggestions.

# Describe alternatives you've considered

Accept the fact that Cataclysm is a zombie survival game. Never.

# Additional context
The dragonfly rework is already up as PR #44779 , as well as a followup draft to liven up swamps a bit ( #44997 ). Feel free to discuss both + the ideas above and make suggestions.

**Comment by SunshineDistillery** [comment]
Winged bugs should make way more noise. You could change them to their own species, give them different move noises, and add the flag for noisy movement. You could also give them a parrot attack to make buzzing sounds.

Not every bug has to be a threat or super interesting. Flies, moths, fireflies, butterflies, could be peaceful animals. 

You can mimic cocoons by making immobile pacifist creatures with no sense tags. 

The ant Queen's hardcoded attacks are really dumb and easily exploitable for infinite food. Any replacement would be an improvement.

**Comment by Venera3** [comment]
On point with the noise, I'll look into how that works. Ants didn't make the list yet because on the one hand they have some very aged gamey stuff going on, on the other they are the only species who are properly fleshed out and not interchangeable with the rest (they get their own mapgen stuff etc.). 
Coccoons are an idea I played with, but its hard to not make them into chitin pinatas, and placing them in a upgrade path is a bit fiddly. I might go for independently spawning "Schrödinger's cocoons", which can hatch into any one species from a list.

**Comment by SunshineDistillery** [comment]
You could make cocoons their own species so other monsters dont attack. For players they would be a resource though.

A problem with random cocoons/eggs is if you put a bunch in one spot, they might hatch as different types.

I made an insect mod a long time ago. I think it's in goatgod's mod pack still. If there's anything in there you want to salvage from the jsons, feel free. There's a speech file with like a hundred variations of 'buzzz' for example.

**Comment by Xpyder** [comment]
I actually recently posted a similar idea [in the forums](https://discourse.cataclysmdda.org/t/zombie-insect-interaction-ideas/25037/2).  For convenience I'll bring over the most relevant parts

As mentioned in #15372 there’s some potential for interesting spider / Wasp interaction with zombies. The core issue most people seem to have is they ignore each other. This was originally implemented because zombies would always eventually win out (because of the rezzing) and kill off the insects which left for uninteresting gameplay.

For spiders (and variants) if they win they could cocoon them and drag them back to the web.
It would eventually feed on it, and if that (or an old cocoon) was opened it could leave behind a “desiccated body” that won’t rez. If a fresh cocoon was opened it could spawn a “dissolving zombie” that would quickly lose health over time and can’t rez. More complicated but also more interesting . This would also provide a hint that there’s more to what’s going on and a connection between them to be investigated

For wasps I like the Emerald Wasp idea, maybe if they win they could infect the zombie so it rezzes as a “dazed” zombie that has no attack or flee instinct. It would slowly take damage over the course of a few days and if allowed to die would spawn a wasp larva / cocoon / adult wasp depending on how complicated their currently implemented life cycle is. To make it more interesting any nearby wasp could lead a dazed zombie back to the nest (if within a certain range) like what the Emerald Cockroach Wasp does. We could even update the wasp description to more closely match it with an “iridescent emerald green body”

**Comment by Xpyder** [comment]
If we do go with the Emerald Wasp idea, it might be good to give them a cooldown on the egg laying (maybe 1-2 per day per wasp) so that if a lone wasp comes across a horde of dead zombies that you cleared they don't all instantly become hosts, that it's a more gradual change

**Comment by Xpyder** [comment]
As for flies what about a mix of giant flies and slime infected flies (like bloaters that explode into poison gas, or disease carriers that spread plague and need environmental protective armor more than cut/bash protection)

**Comment by Xpyder** [comment]
For cocoons you might have them spawn pupating jelly instead of chitin, maybe give that a low level usefulness on its own (maybe moderate calories, but with toxins, low joy, and very mild mutative properties (the last would need to be balanced so it doesn't become an easy mutagen replacement for those with robust genetics)

**Comment by SunshineDistillery** [comment]
I think venera is working in the jsons. Altering corpses would require code changes. I dont think

**Comment by Venera3** [comment]
Yeah, I'm still coming to grips with what's possible with the spell stuff. Accelerating rot is in there I think, and would work to make flies interactive.

Coccooned corpses for spiders is def a good idea, I wanted to make those forest patches interesting anyway. I could implement them with two indistinguishable variants,  one with a corpse+loot and one with a zombie inside ( it would probably work as a hacky pseudoegg corpse that rots into zeds ).

The normal wasps I thik I'd leave as a social species cause we need a more metal foil to the nice and friendly bees, and the fun parasitic wasps are solitary. Dermatiks on the other hand would work, if they can add their effect to mobs.

**Comment by SunshineDistillery** [comment]
Map extras could place special corpses, but there's no way for real time cocooning.

Map extras are quirky. For one, they dont choose among all entries with an id. They just choose the first one. So you can make a blank extra that places a nested mapgen. Then make different chunks for it to place. That way one mapextra will have different results.

There are hard coded eggsacks already that spit out baby spiders. Maybe some one would convert it to a json option for you. An examine action for furniture that summons x monster.

Good luck.

**Comment by LyleSY** [comment]
Broadly love the idea, and I like fireflies as a new thing. I'm not sure it makes sense for flies to mutate into fireflies. What I find interesting about flies is their symbolic connection to plague and pestilence and the horror Fly movies where people become flies and flies become people. I've gotten a lot of negative feedback about mutant spiders that they're too easy at every evolutionary step. I've also gotten negative feedback that there's no reason to fight mutant insects other than just to fight them. The meat is useless now though the chitin can be helpful for farming. Something I've put into DinoMod to give some reward to players is to give some kinds of dinos a chance of having something interesting in their stomachs like a human body or a gun or some science equipment. I'm not sure if that's too gamey for the base game, but I thought it was fun. They could also have stuff stuck to them that they drop when they're killed that has some small value. The new venom system might solve all this though.

**Comment by Venera3** [comment]
I'm making fireflies their own thing after some thought.

The whole idea of redesigning insects came after I saw how uninteresting
they were as I added the venom stuff. I think making the player engage with
stuff for loot is generally a good thing, but for the base game you'd need
some really good arguments.
Other than that, well predators are hunting you, so they'll engage with the
player, and flies are all right as prey for some incidental
insect-on-insect violence to behold/scavenge after.

**Comment by Xpyder** [comment]
> There are hard coded eggsacks already that spit out baby spiders. Maybe some one would convert it to a json option for you. An examine action for furniture that summons x monster.

I don't mind digging into some of the cpp files.  My main talent is in Rust an Powershell so I have trouble making extensive or new changes (a lot of my changes copy approaches from other places and adapt them rather than creating fresh)

If there are some examples of how to convert this stuff to json (like past PRs that have been merged) and where to find it (unique phrases, or filenames and line numbers) I'm happy to put some work into it.  I don't know that I'll be able to complete it by myself but I might be able to make it easier for someone who can.

**Comment by Xpyder** [comment]
> The normal wasps I think I'd leave as a social species cause we need a more metal foil to the nice and friendly bees, and the fun parasitic wasps are solitary. Dermatiks on the other hand would work, if they can add their effect to mobs.

This still leaves some potential issues if the zombies are rezzing and pushing into the wasp nests.  I suppose they could just eat the dead bodies (many wasps are carnivorous).  For another example of a wasp that could be used as a model there's [Glyptapanteles](https://en.wikipedia.org/wiki/Glyptapanteles).  There's some [evidence](https://www.sciencedirect.com/topics/agricultural-and-biological-sciences/glyptapanteles) of them being gregarious rather than solitary but I'm unclear if that's just the larval stage.  Further reading [here](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6881475/) (this page may be slow to load)

**Comment by Venera3** [comment]
Xpyder, if you're up for some lightish (based on my complete lack of understanding, mind) coding I could use your help in making the flying monsters be able to have footsteps. There's an explicit function in monmove.cpp disallowing that, but either there's more going on or I messed up in the testing cause changing that still didn't work for me and I don't trust myself to do anything more than change a variable in there.

**Comment by Resok** [comment]
Any plans to change spider venom to paralytic? Most spiders have some form of neurotoxin that attacks the nervous system and is primarily a paralytic venom. A few spider breeds have necrotic venoms but it's far less common compared to the various intensities of paralytic neurotoxins.

Among the species that have neurotoxins are - web weavers (black widows, orb weavers, etc.), wolf spiders, jumping spiders and trapdoor spiders.

**Comment by Venera3** [comment]
Yes, when I get around to redoing them properly. As opposed to wasps, centipedes and the like spiders generally aren't really physically imposing, so their venom should do most of their work for them. The current venom implementation is mostly to include the more common critters in my "involuntary playtester recruitment drive" and to quickly make them a bit scarier. 
We'll see how long that second pass takes with 0F (and content freeze) coming and the dark days ahead IRL.

**Comment by kevingranade** [comment]
This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/gas-sacs-and-why-no-they-wouldnt-work/25566/13

**Comment by actual-nh** [comment]
It has just occurred to me, while reading the PMC link @Venera3 gave above (thanks!), that the blob's behavior in simply reusing what's there may well extend to reusing things already known to it from other species (particularly other species it is altering) - note mutagen responding to species it's exposed to. This would seem especially likely in species that are exposed to each other - e.g., predator/prey, parasite/host... so dermatiks that have grown inside, say, paper wasps may well have paper-building themselves.

**Comment by slimeboy460** [comment]
Me and @Venera3 were discussing how we could simulate corpse consumption in my #47013 PR (and PMs). As well as this I also had a look at the giant insect overhaul.

I saw that I-Am-Erk wanted to have a beetle which rolled corpses like dung beetles roll poop. He might actually be in luck because there is a sub-species of beetles called dermestid. Although i am currently designing my beetles based on the good ol’ stag beetle, I could probably make a variant for this species.

**Comment by actual-nh** [comment]
BTW, why is the Firefly species INSECT, not INSECT_FLYING?

**Comment by Venera3** [comment]
> BTW, why is the Firefly species INSECT, not INSECT_FLYING?

Whooops. Nice catch, thank you!

**Comment by actual-nh** [comment]
This [CDDA browser](http://cdda-trunk.chezzo.com/monsters/species/INSECT) can be useful - you can list monsters by species.

**Comment by Venera3** [comment]
Opened a Discussion to keep this a bit more visible, I'll keep the issue open as a reference point for future PRs.

**Comment by kevingranade** [comment]
This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/monster-brainstorming/27087/5
